### PR TITLE
Using Weather Data to update the Todays Weather Page

### PR DESCRIPTION
**Pull request by iAmTheDolphin**

### CODE REVIEW

File: PackingList/app/src/main/res/layout/activity_todays_weather.xml [hunk 1]
```diff
@@ -18,11 +18,18 @@
 
         <TextView
             android:id="@+id/locationTextView"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
```
Comment (andrasveto) [inline PackingList/app/src/main/res/layout/activity_todays_weather.xml:22]: Why were these changed? We should avoid using absolute values for constraints wherever possible

File: PackingList/app/src/main/res/layout/activity_todays_weather.xml [hunk 2]
```diff
@@ -18,11 +18,18 @@
...
             android:gravity="center"
-            android:textSize="24sp"
-            android:textColor="@android:color/black" />
+            android:textColor="@android:color/black"
+            android:textSize="24sp" />
+
+        <TextView
+            android:id="@+id/longitudeTextView"
+            android:layout_width="match_parent"
+            android:layout_height="45dp"
```
Comment (andrasveto) [inline PackingList/app/src/main/res/layout/activity_todays_weather.xml:30]:
```suggestion
            android:layout_height="wrap_content"
```

File: PackingList/app/src/main/java/com/example/packinglist/TodaysWeatherActivity.kt [hunk 3]
```diff
@@ -1,37 +1,88 @@
...
+            "sleet" -> return WeatherState.Snow
+            "wind" -> return WeatherState.Clouds
+            "fog" -> return WeatherState.Clouds
+            "cloudy" -> return WeatherState.Clouds
+            "partly-cloudy-day" -> return WeatherState.Clouds
+            "partly-cloudy-night" -> return WeatherState.Clouds
+            else -> {
+                // for the ones that they will implement in the future
+                // https://darksky.net/dev/docs#data-point
+                return WeatherState.Dev
```
Comment (andrasveto) [inline PackingList/app/src/main/java/com/example/packinglist/TodaysWeatherActivity.kt:40]: Let's make the function return `WeatherState?` and return `null` here instead so that we can remove `WeatherState.Dev` from the enum.

Also, if we get nuanced weather data like this then it would be nice to use that to increase the granularity of our recommendations by adding more states to the enum.

Comment (iAmTheDolphin) [reply PackingList/app/src/main/java/com/example/packinglist/TodaysWeatherActivity.kt]: returns a WeatherState? now, and weatherData handles it if null.

I thought so too on the nuance. We don't have time for it today, but we could make our app very detailed by weather

File: PackingList/app/src/main/res/layout/activity_todays_weather.xml (hunk 2 above)
Comment (iAmTheDolphin) [reply PackingList/app/src/main/res/layout/activity_todays_weather.xml]: fixed

File: PackingList/app/src/main/res/layout/activity_todays_weather.xml (hunk 1 above)
Comment (iAmTheDolphin) [reply PackingList/app/src/main/res/layout/activity_todays_weather.xml]: fixed
